pkg/registry/generic/etcd/etcd.go

```diff
@@ -427,7 +430,15 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 			if lastGraceful > 0 {
 				return out, nil
 			}
-			// fall through and delete immediately
+			// If we are here, the registry supports grace period mechanism and
+			// we are intentionally delete gracelessly. In this case, we may
+			// enter a race with other k8s components. If other component wins
+			// the race, the object will not be found, and we should tolerate
+			// the NotFound error. See
+			// https://github.com/kubernetes/kubernetes/issues/19403 for
+			// details.
+			ignoreNotFound = true
```

Comment:
@liggitt I keep var name as `ignoreNotFound` as I think `returnLastExistingForNotFound` is redundant. I could change if you insist.